src/jit/codegenxarch.cpp

```diff
@@ -5172,16 +5172,17 @@ void CodeGen::genTransferRegGCState(regNumber dst, regNumber src)
 //     pass in 'addr' for a relative call or 'base' for a indirect register call
 //     methHnd - optional, only used for pretty printing 
 //     retSize - emitter type of return for GC purposes, should be EA_BYREF, EA_GCREF, or EA_PTRSIZE(not GC)
```

Comment:
I see another gc-hole in genEmitGSCookieCheck() that doesn't handle gc-ness of struct return in two registers. 

GS cookie check is emitted after the codegen of GT_RETURN or a tail call or jmp method. It not considered part of epilog.  Hence genEmitGSCookieCheck() explicitly checks to see if the return type is REF and if so keeps Return Register alive for the duration of GS cookie check.  Say if the method is returning a struct in two return registers, this logic needs to be updated to explicitly keep RAX and/or RDX alive for the duration of GS cookie check.